languages/c/reference/exercise-concepts/two-fer.md

```diff
@@ -0,0 +1,22 @@
+# Concepts of two-fer
+
+```c
+#include "two_fer.h"
+#include <stdio.h>
+
+void two_fer(char *buffer, const char *name) {
+    sprintf(buffer, "One for %s, one for me.", name ? name : "you");
+}
+```
+
+- [Header Files](https://www.gnu.org/software/libc/manual/html_mono/libc.html#Header-Files): used to access the standard library
+- [C Standard Library header files](https://en.cppreference.com/w/c/header): stdio.h is a C standard library header file
+- [Function Definitions](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Function-Definitions): used as the main entry point for the exercise
+- [Function Parameters](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Function-Parameters): buffer and name are passed as arguments
+- [Integer Types](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Integer-Types): buffer and name are integer types
+- [Pointers](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Pointers): buffer and name are pointers
+- [Type Qualifiers](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Type-Qualifiers): name is const
+- [Calling Functions](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Calling-Functions): the sprintf function is called
+- [sprintf](https://en.cppreference.com/w/c/io/fprintf): the sprintf function is used
```

Comment:
This can probably include more concepts, e.g. string formatting, format specifiers, buffer and name as arguments here, etc

Reply:
I think `sprintf` is an implementation of a more general Concept. That general concept is "string formatting" I think.

Reply:
Yeah, I'm not sure what the best concept would be for different functions in the C standard library. Should we have a [Strings library](https://en.cppreference.com/w/c/string) concept, or would that be too big?

If we have a "string formatting" concept, would that include input/output functions such as `printf` and `scanf`?

Reply:
> If we have a "string formatting" concept, would that include input/output functions such as printf and scanf?

Hmmm, tough choice. They are both console input/output functions, right? As such, might be hard to write test for them. @iHiD any thoughts?

Reply:
> Should we have a Strings library concept, or would that be too big?

Maybe. It depends a bit on if there are different concepts in the string library. For example, I'd consider finding the index of a character a part of the "Strings" concept. In C#, I also included basic "String formatting" functionality in the "Strings" exercise, but it depends on the language if that makes sense. My reason for including it in C# is because so many exercises will depend on it later and as it is very common in C# to format strings.

Reply:
> Hmmm, tough choice. They are both console input/output functions, right? As such, might be hard to write test for them.

Yes, you're right, it's probably best to skip those. I thought `sprintf` was a part of the "strings library", but it's actually a part of "File input/output". If you know `sprintf` you'll have no trouble understanding `printf`.

I guess we'll figure out how to divide the C standard library function when we start thinking about what should be included in each concept.

Reply:
My guidance for all this is to think "by teaching X, will someone then know how to find Y themselves". If, after learning X, the existance of Y is not obvious, or where to go to read it is not obvious, then Y is probably something best taught explicitly. 

So for example, in Ruby, in one exercise, we'd teach things like `string.upcase` and `string.substring` and `string.replace` all of which are methods of strings and appear on the same docs page. So I'm happy to say all of those fit into the same sort of bucket of `string-methods` (or something named better). However, `sprintf` is defined separately and is called in a functional manner, so I wouldn't think this would be implicitly included in this `string-methods` concept, so I'd either explicitly mention it in the concept-exercise (maybe in the after.md) or put it as part of a separate concept.

So in my mind, the grouping is largely based on how the user will find/group/understand things in their mental scaffolding. Is that useful in any way?